src/app/notifications/broker/events/notifications-broker-events.component.ts

```diff
@@ -0,0 +1,464 @@
+    this.paginationService.getFindListOptions(this.paginationConfig.id, this.defaultConfig).pipe(
+      distinctUntilChanged(),
+      switchMap((options: FindListOptions) => this.notificationsBrokerEventRestService.getEventsByTopic(
+        this.topic,
+        options,
+        followLink('target'), followLink('related')
+      )),
+      getFirstCompletedRemoteData(),
+    ).subscribe((rd: RemoteData<PaginatedList<NotificationsBrokerEventObject>>) => {
+      if (rd.hasSucceeded) {
+        this.isEventLoading.next(false);
+        this.totalElements$ = observableOf(rd.payload.totalElements);
+        this.setEventUpdated(rd.payload.page);
+      } else {
+        throw new Error('Can\'t retrieve Notifications Broker events from the Broker events REST service');
+      }
+      this.notificationsBrokerEventRestService.clearFindByTopicRequests();
+    });
+  }
+
+  /**
+   * Unsubscribe from all subscriptions.
+   */
+  ngOnDestroy(): void {
+    this.subs
+      .filter((sub) => hasValue(sub))
+      .forEach((sub) => sub.unsubscribe());
+  }
+
+  /**
+   * Set the project status for the Notifications Broker events.
+   *
+   * @param {NotificationsBrokerEventObject[]} events
+   *    the Notifications Broker event item
+   */
+  protected setEventUpdated(events: NotificationsBrokerEventObject[]): void {
+    this.subs.push(
+      from(events).pipe(
+        mergeMap((event: NotificationsBrokerEventObject) => {
+          const related$ = event.related.pipe(
+            getFirstCompletedRemoteData(),
+          );
+          const target$ = event.target.pipe(
+            getFirstCompletedRemoteData()
+          );
+          return combineLatest([related$, target$]).pipe(
+            map(([relatedItemRD, targetItemRD]: [RemoteData<Item>, RemoteData<Item>]) => {
+              const data: NotificationsBrokerEventData = {
+                event: event,
+                id: event.id,
+                title: event.title,
+                hasProject: false,
+                projectTitle: null,
+                projectId: null,
+                handle: null,
+                reason: null,
+                isRunning: false,
+                target: (targetItemRD?.hasSucceeded) ? targetItemRD.payload : null,
+              };
+              if (relatedItemRD?.hasSucceeded && relatedItemRD?.payload?.id) {
+                data.hasProject = true;
+                data.projectTitle = event.message.title;
+                data.projectId = relatedItemRD?.payload?.id;
+                data.handle = relatedItemRD?.payload?.handle;
+              }
+              return data;
+            })
+          );
+        }),
+        scan((acc: any, value: any) => [...acc, value], []),
+        take(events.length)
+      ).subscribe(
+        (eventsReduced) => {
+          this.eventsUpdated$.next(eventsReduced);
+        }
+      )
+    );
+  }
+
+  protected computePIDHref(event: OpenaireBrokerEventMessageObject) {
```

Comment:
This method has many hardcoded URL prefixes which sites may need to occasionally change.  I'm wondering if there's a way to move this logic to the backend where it could be more easily configurable?  Is there a reason why the prefix needs to be added here in the UI, instead of just making sure the full URL can be sent by the backend?